add deprecation chapter

MrTango opened the pull request:
## First-time contributors

You **must** read and follow our [First-time contributors](https://6.docs.plone.org/contributing/first-time.html).

---

## Submit a pull request

Thank you for your contribution to the Plone Documentation.

Before submitting this pull request, please make sure you follow our guides:

-   [Contributing to Plone documentation](https://6.docs.plone.org/contributing/index.html)
-   [Building and checking the quality of documentation](https://6.docs.plone.org/contributing/setup-build.html)
-   [Authors guide](https://6.docs.plone.org/contributing/authors.html)
-   [MyST reference](https://6.docs.plone.org/contributing/myst-reference.html)

## Issue number

- Fixes #

## Description

_Write a description of the fixes or improvements._

## Add screenshots or links to a preview of the changes



<!-- readthedocs-preview plone6 start -->
----
📚 Documentation preview 📚: 

- https://plone6--1850.org.readthedocs.build/developer-guide/index.html
- https://plone6--1850.org.readthedocs.build/developer-guide/deprecation.html
- https://plone6--1850.org.readthedocs.build/conceptual-guides/index.html
- https://plone6--1850.org.readthedocs.build/conceptual-guides/deprecation.html
- 

<!-- readthedocs-preview plone6 end -->

MrTango committed:
add deprecation chapter

ale-rt commented on docs/backend/deprecation.md:
```diff
@@ -0,0 +1,311 @@
+While refactoring code, moving modules, functions, classes and methods is often needed.
+To not break third party code imports from the old place or usage of old functions/ methods must work for while.
+Deprecated methods are usually removed with the next major release of Plone.
+Following the [semantic versioning guideline](http://semver.org) is recommended.
```
```suggestion

Following the [semantic versioning guideline](http://semver.org) is recommended.
```

ale-rt commented on docs/backend/deprecation.md:
```diff
@@ -0,0 +1,311 @@
+
+### Deprecating a template position
+
+Sometimes we need to move templates to new locations. Since addons often use jbot to override templates by their position, we need to point them to the new position as well as make sure that the override still works with the old position.
```
```suggestion
Sometimes we need to move templates to new locations. Since addons often use [z3c.jbot](https://github.com/zopefoundation/z3c.jbot) to override templates by their position, we need to point them to the new position as well as make sure that the override still works with the old position.
```

ale-rt commented on docs/backend/deprecation.md:
```diff
@@ -0,0 +1,311 @@
+
+Sometimes we need to move templates to new locations. Since addons often use jbot to override templates by their position, we need to point them to the new position as well as make sure that the override still works with the old position.
+
+
```
```suggestion

To deprecate a package:

```

ale-rt commented on docs/backend/deprecation.md:
```diff
@@ -0,0 +1,311 @@
+Sometimes we need to move templates to new locations. Since addons often use jbot to override templates by their position, we need to point them to the new position as well as make sure that the override still works with the old position.
+
+
+In the old package folders `__init__.py` we add a mapping dict with the name `jbot_deprecations` of old and new template location:
```
```suggestion
1. In the old package folders `__init__.py` add a dictionary `jbot_deprecations` that maps the old template locations to their new counterparts, e.g.:
```

ale-rt commented on docs/backend/deprecation.md:
````diff
@@ -0,0 +1,311 @@
+}
+```
+
+add a jbot-deprecations config to the `configure.zcml` file
````
```suggestion
2. Add this deprecation snippet to the package `configure.zcml` file:
```

ale-rt commented on docs/backend/deprecation.md:
```diff
@@ -0,0 +1,311 @@
+
+add a jbot-deprecations config to the `configure.zcml` file
+
+create a `jbot_deprecations.zcml` file:
+
```
```suggestion
```

ale-rt commented on docs/backend/deprecation.md:
```diff
@@ -0,0 +1,311 @@
+    xmlns="http://namespaces.zope.org/zope"
+    xmlns="http://namespaces.zope.org/zope"
+    xmlns:browser="http://namespaces.zope.org/browser"
+    xmlns:browser="http://namespaces.zope.org/browser"
```
```suggestion
```

ale-rt commented on docs/backend/deprecation.md:
```diff
@@ -0,0 +1,311 @@
+
+<configure
+    xmlns="http://namespaces.zope.org/zope"
+    xmlns="http://namespaces.zope.org/zope"
```
```suggestion
```

MrTango committed:
fix broken links in deprecation chapter

ale-rt commented on docs/backend/deprecation.md:
````diff
@@ -0,0 +1,311 @@
+</configure>
+```
+
+this will load the defined `jbot_deprecations` dictionary.
````
```suggestion
If a `z3c.jbot` version that supports deprecation is found, trying to override the template with the old location will trigger a deprecation warning that will instruct the user to rename its override file.
```

ale-rt commented on docs/backend/deprecation.md:
```diff
@@ -0,0 +1,311 @@
+    xmlns:zcml="http://namespaces.zope.org/zcml"
+    >
+
+  <configure zcml:condition="have jbot-deprecations">
+    <include package="z3c.jbot" file="meta.zcml" />
+    <browser:jbotDeprecated dictionary=".jbot_deprecations" />
+  </configure>
```
```suggestion
  <include
      zcml:condition="installed z3c.jbot"
      package="z3c.jbot" 
      file="meta.zcml" 
      />
  <browser:jbotDeprecated
      zcml:condition="have jbot-deprecations"
      dictionary=".jbot_deprecations"
      />
```

ale-rt replied:
I think that the `meta.zcml` has to be loaded before checking the feature. If we have the `jbot-deprecations` feature, it means the meta has been already loaded.

CC @mauritsvanrees

MrTango committed:
Update docs/backend/deprecation.md

Co-authored-by: Alessandro Pisa <alessandro.pisa@gmail.com>

MrTango committed:
Update docs/backend/deprecation.md

Co-authored-by: Alessandro Pisa <alessandro.pisa@gmail.com>

MrTango committed:
Update docs/backend/deprecation.md

Co-authored-by: Alessandro Pisa <alessandro.pisa@gmail.com>

MrTango committed:
Update docs/backend/deprecation.md

Co-authored-by: Alessandro Pisa <alessandro.pisa@gmail.com>

MrTango committed:
Update docs/backend/deprecation.md

Co-authored-by: Alessandro Pisa <alessandro.pisa@gmail.com>

MrTango committed:
Update docs/backend/deprecation.md

Co-authored-by: Alessandro Pisa <alessandro.pisa@gmail.com>

MrTango committed:
Update docs/backend/deprecation.md

Co-authored-by: Alessandro Pisa <alessandro.pisa@gmail.com>

MrTango committed:
Update docs/backend/deprecation.md

Co-authored-by: Alessandro Pisa <alessandro.pisa@gmail.com>

MrTango committed:
Update docs/backend/deprecation.md

Co-authored-by: Alessandro Pisa <alessandro.pisa@gmail.com>

MrTango committed:
Update docs/backend/deprecation.md

Co-authored-by: Alessandro Pisa <alessandro.pisa@gmail.com>

MrTango committed:
remove meta.zcml attribute from z3c.jbot include

MrTango committed:
Update docs/backend/deprecation.md

Co-authored-by: Alessandro Pisa <alessandro.pisa@gmail.com>

ale-rt commented on docs/backend/deprecation.md:
```diff
@@ -0,0 +1,314 @@
+---
+myst:
+  html_meta:
+    "description": "A guide how to do depracations, including Python, ZCML and templates in Plone."
```
```suggestion
    "description": "A guide how to do deprecations, including Python, ZCML and templates in Plone."
```

ale-rt commented on docs/backend/deprecation.md:
```diff
@@ -0,0 +1,314 @@
+myst:
+  html_meta:
+    "description": "A guide how to do depracations, including Python, ZCML and templates in Plone."
+    "property=og:description": "A guide how to do depracations, including Python, ZCML and templates in Plone."
```
```suggestion
    "property=og:description": "A guide how to do deprecations, including Python, ZCML and templates in Plone."
```

ale-rt commented on docs/backend/deprecation.md:
```diff
@@ -0,0 +1,314 @@
+    "description": "A guide how to do depracations, including Python, ZCML and templates in Plone."
+    "property=og:description": "A guide how to do depracations, including Python, ZCML and templates in Plone."
+    "property=og:title": "Deprecation"
+    "keywords": "depracation, zcml, template, jbot"
```
```suggestion
    "keywords": "deprecation, zcml, template, jbot"
```

ale-rt committed:
Fix ty

ale-rt committed:
Fix typo

ale-rt committed:
Fix typo

ale-rt commented on docs/backend/deprecation.md:
```diff
@@ -0,0 +1,314 @@
+
+### Why Deprecation
+
+At some point we
```
```suggestion
At some point we:
```

ale-rt commented on docs/backend/deprecation.md:
```diff
@@ -0,0 +1,314 @@
+
+Moving a module, class, function, etc to another place
+
+: For some reason, i.e. merging packages, consistent api or resolving cirular import problems, we need to move code around.
```
```suggestion
: For some reason, i.e. merging packages, consistent API or resolving cirular import problems, we need to move code around.
```

ale-rt commented on docs/backend/deprecation.md:
```diff
@@ -0,0 +1,314 @@
+
+Renaming
+
+: We may want to rename classes, methods, functions or global or class variables in order to get a more consistent api or because of a typo, etc.
```
```suggestion
: We may want to rename classes, methods, functions or global or class variables in order to get a more consistent API or because of a typo, etc.
```

ale-rt commented on docs/backend/deprecation.md:
```diff
@@ -0,0 +1,314 @@
+At some point we
+
+- need to get rid of old code,
+- want to unify api style (consistent api),
```
```suggestion
- want to unify API style (consistent API),
```

ale-rt replied:
I saw below in other parts of the document that `API`, was uppercased.
For consistency I would always use it uppercased.
I also think it is more correct to have it uppercased.

yurj commented on docs/backend/deprecation.md:
````diff
@@ -0,0 +1,314 @@
+...
+```
+
+This just sets a configuration option in `zope.conf`.
````
For non buildout users, could be interesting to have the zope.conf snippet available here, if possible.

ale-rt replied:
~~~suggestion
This adds this line to the `zope.conf` file:

```
debug-mode on
```
~~~

Or maybe add these lines above and then say that buildout user can have this with the proper recipe settings.

MrTango committed:
Update docs/backend/deprecation.md

Co-authored-by: Alessandro Pisa <alessandro.pisa@gmail.com>

MrTango committed:
Update docs/backend/deprecation.md

Co-authored-by: Alessandro Pisa <alessandro.pisa@gmail.com>

MrTango committed:
Update docs/backend/deprecation.md

Co-authored-by: Alessandro Pisa <alessandro.pisa@gmail.com>

MrTango committed:
Update docs/backend/deprecation.md

Co-authored-by: Alessandro Pisa <alessandro.pisa@gmail.com>

MrTango committed:
Update docs/backend/deprecation.md

Co-authored-by: Alessandro Pisa <alessandro.pisa@gmail.com>

MrTango committed:
Merge branch '6.0' into chapter-deprecation

stevepiercy commented on .gitignore:
```diff
@@ -13,3 +13,6 @@ docs/volto
 
 # editor files
 .vscode
+/bin/
+/lib
+/lib64
```
Documentation does not generate these directories, and they are not editor files. Why did you add them?

ale-rt replied:
```suggestion
```

jensens commented on docs/backend/deprecation.md:
```diff
@@ -0,0 +1,318 @@
+
+Deprecation of a whole package
+
+: A whole package (folder with `__init__.py`)
```
```suggestion
: A whole [package](https://docs.python.org/3/tutorial/modules.html#packages)
```

jensens commented on docs/backend/deprecation.md:
```diff
@@ -0,0 +1,318 @@
+  - all imports still working, logging deprecation warnings on first import
+  - ZCML still exists, but is empty (or includes the zcml from the new place if theres no auto import (i.e. for meta.zcml).
+
+Deprecation of a whole python egg
```
```suggestion
Deprecation of a whole released/ installable package.
```

stevepiercy commented on docs/backend/deprecation.md:
```diff
@@ -0,0 +1,318 @@
+
+: They may got renamed for consistency or are superfluos after an update.
+  Code does not need to break to support this.
+
```
Everything up to this point is a Conceptual Guide, and needs to be split into a separate file placed in the `conceptual-guides` directory.

Everything after this point is a How-to Guide and should be moved under  [Developer guide](https://6.docs.plone.org/developer-guide/index.html). This section is a recent addition, and more suitable for this content.

stevepiercy reviewed (changes requested):
Please follow the amended original plan https://github.com/plone/documentation/issues/1464#issuecomment-2442579972

jensens reviewed (changes requested):
Overall fine, found two minor improvements.

ale-rt commented on docs/backend/deprecation.md:
```diff
@@ -0,0 +1,318 @@
+To not break third party code imports from the old place or usage of old functions/ methods must work for while.
+Deprecated methods are usually removed with the next major release of Plone.
+
+Following the [semantic versioning guideline](http://semver.org) is recommended.
```
```suggestion
Following the [semantic versioning guideline](https://semver.org) is recommended.
```

ale-rt commented on docs/backend/deprecation.md:
```diff
@@ -0,0 +1,318 @@
+
+### Help Programmers, No annoyance
+
+Deprecation has to support the consumers of the code - the programmers using it.
```
```suggestion
The developers should use code deprecations to support the consumers of the code.
```

stevepiercy committed:
Apply suggestions from code review

Co-authored-by: Alessandro Pisa <alessandro.pisa@gmail.com>
Co-authored-by: Jens W. Klein <jk@kleinundpartner.at>

stevepiercy committed:
Merge branch '6.0' into chapter-deprecation

stevepiercy committed:
Merge branch '6.0' into chapter-deprecation

sneridagh commented on docs/backend/index.md:
```diff
@@ -22,6 +22,7 @@ behaviors
 configuration-registry
 content-types/index
 control-panels
+deprecation
```
Do you all think that this insertion point in the TOC is the most appropiate place for this subject?

stevepiercy replied:
It is not. See https://github.com/plone/documentation/pull/1850#pullrequestreview-2614132745.

jensens committed:
Merge branch '6.0' into chapter-deprecation

jensens committed:
Merge branch '6.0' into chapter-deprecation

ale-rt committed:
Merge branch '6.0' into chapter-deprecation

MrTango committed:
Merge branch '6.0' into chapter-deprecation

stevepiercy commented:
@MrTango I'm awaiting changes per https://github.com/plone/documentation/pull/1850#pullrequestreview-2614132745 before merging. Please let me know. Thank you!

stevepiercy commented:
@MrTango should I close this PR as abandoned and won't fix? It needs help before it can be merged.

ale-rt commented:
> @MrTango should I close this PR as abandoned and won't fix? It needs help before it can be merged.

It would be a pity because it is a very nice write up :/

stevepiercy commented:
> It would be a pity because it is a very nice write up :/

Indeed. However, my review has not been addressed. I don't know why. No response from @MrTango implies this PR is not needed. If you care to push it forward, please be my guest.

jensens commented:
> @MrTango should I close this PR as abandoned and won't fix? It needs help before it can be merged.

I will take a look. I wrote this initially back in the days... so I would like to see it survive.

jensens committed:
Merge branch '6.0' into chapter-deprecation

jensens committed:
split deprecation guide into two file according to diataxis principles

jensens commented:
Ok, this should work. @stevepiercy what do you think?

ale-rt commented on docs/developer-guide/deprecation.md:
````diff
@@ -0,0 +1,255 @@
+2. Add this deprecation snippet to the package `configure.zcml` file:
+
+```{code-block} xml
+:emphasize-lines: 6,9-12
````
See https://plone6--1850.org.readthedocs.build/developer-guide/deprecation.html#deprecating-a-template-position:

<img width="1007" height="403" alt="Image" src="https://github.com/user-attachments/assets/79ed25c2-9b06-4480-b227-203a2f255755" />

```suggestion
:emphasize-lines: 7-14
```

jensens committed:
Update docs/developer-guide/deprecation.md

Co-authored-by: Alessandro Pisa <alessandro.pisa@gmail.com>

ale-rt reviewed (dismissed):
Thank you very much to all of you :)
@stevepiercy do you mind having another look?

stevepiercy committed:
Merge branch '6.0' into chapter-deprecation

stevepiercy committed:
- Fix English grammar and syntax
- Generalize for npm and Volto
- Clean up MyST style and formatting

stevepiercy committed:
Refine some grammar

stevepiercy committed:
- Add MyST syntax to link to Python objects, improve rendering
- Clean up grammar, MyST syntax
- Indent code within list items.
- Use ruff style code example formatting.

stevepiercy reviewed (changes requested):
This is really good. I can tell because even I can understand it.

I have only one question that needs to be addressed. Thank you!

jensens committed:
Merge branch '6.0' into chapter-deprecation

jensens commented:
> This is really good. I can tell because even I can understand it.
> 
> I have only one question that needs to be addressed. Thank you!

sorry, I can not find the question...

stevepiercy commented on docs/developer-guide/deprecation.md:
```diff
@@ -0,0 +1,263 @@
+
+### Zope
+
+Zope configures logging and warnings, so the steps as described below in {ref}`deprecation-warning-python-label` aren't needed.
```
If the steps in the next section aren't needed, then why do we include them?

jensens replied:
With next step you refer to `debug-mode on`? IIRC it is needed to show the messages. I wrote this down years ago, so this may have been subject to change.

jensens replied:
And with `cookiecutter-zope-instance` based installs we only need to use the PYTHONWARNINGS variable or start the interpreter with parameters (like below in Python).

Thinking further, I am not 100% sure if the above method is anyway ZServer only?
Does anybody in here know more details?

stevepiercy replied:
@jensens you might be able to see it more clearly on the pull request preview, beginning at https://plone6--1850.org.readthedocs.build/developer-guide/deprecation.html#zope.

This sentence refers to the next section entitled [Python](https://plone6--1850.org.readthedocs.build/developer-guide/deprecation.html#deprecation-warning-python-label), and it states that the steps in that section are not needed.

jensens replied:
We can drop the whole section `Zope` with plone.recipe.zopeinstance since this feature was dropped with WSGI server usage. So, it does not make sense to keep the Python subheading, right?
See last bullet point at https://github.com/plone/plone.recipe.zope2instance/blob/0452e05386708564b17ae1de1aafe708a4d88ccb/CHANGES.rst#610-2018-12-28 

Shall we mention somehow this change in the developer guide or focus on the current state?

stevepiercy replied:
I wish I could offer advice, but this was before my time. I'll go with what you think is true and accurate, and to the best of your knowledge. We can also wait a few days for other old timers to offer advice. The only thing I can refer to is this search result:

https://6.docs.plone.org/search.html?q=plone.recipe.zopeinstance

I'd focus on the current state of Plone 6.x.x. I'd use a `{versionchanged}` or `{versionremoved}` MyST directive, but we should use Plone 6.x.x or 5.x.x as the referred version, whenever the package got released with Plone, not the package's version. The content of that directive should include whatever is helpful to Plone 6 developers for what has changed. There might even be some references in the [Upgrade Guide](https://6.docs.plone.org/backend/upgrading/index.html). Why, yes, there is: https://6.docs.plone.org/backend/upgrading/version-specific-migration/upgrade-to-52.html#wsgi

Does that help at all?

mauritsvanrees replied:
> We can drop the whole section `Zope` with plone.recipe.zopeinstance since this feature was dropped with WSGI server usage. So, it does not make sense to keep the Python subheading, right?

I think you mean we don't need to keep the *Zope* subheading.  The *Python* subheading still seems accurate.

I must say I *always* get deprecation warnings when starting Plone though, even though I don't think I have configured anything to enable it.  So when I don't want them, I explicitly *disable* them with `export PYTHONWARNINGS=ignore`.  This may depend on how Python is installed, and with which flags it was built.

> Shall we mention somehow this change in the developer guide or focus on the current state?

Indeed focus on current state, but if possible mention the change as Steve says.

jensens replied:
We then have one subheading, that is a bit pointless.  But if it helps, ok.

stevepiercy replied:
There would be two, one for Python and the other for running tests.

stevepiercy reviewed (commented):
@jensens I thought I had submitted it. Here ya go!

jensens commented on docs/developer-guide/deprecation.md:
```diff
@@ -0,0 +1,263 @@
+    
+    Output can be enabled by starting the Python interpreter with the {ref}`-W[all|module|once]  <python:using-on-warnings>` argument.
+ 
+    It's possible to enable output in code, too.
```
We are missing an important way of control here:

```suggestion
    Output can be enabled by starting the Python interpreter with the {ref}`-W[all|module|once]  <python:using-on-warnings>` argument.
    
    As an alternative, the environment variable {envvar}`python:PYTHONWARNINGS` can be set to `default`, in other words, `PYTHONWANRINGS=default`.
 
    It's possible to enable output in code, too.
```

stevepiercy replied:
Amended to use a link to the Python object.

stevepiercy committed:
Update docs/developer-guide/deprecation.md

Co-authored-by: Jens W. Klein <jk@kleinundpartner.at>

stevepiercy committed:
Merge branch '6.0' into chapter-deprecation

mauritsvanrees commented on docs/developer-guide/deprecation.md:
```diff
@@ -0,0 +1,265 @@
+    
+    Output can be enabled by starting the Python interpreter with the {ref}`-W[all|module|once]  <python:using-on-warnings>` argument.
+    
+    As an alternative, the environment variable {envvar}`python:PYTHONWARNINGS` can be set to `default`, in other words, `PYTHONWANRINGS=default`.
```
typo

```suggestion
    As an alternative, the environment variable {envvar}`python:PYTHONWARNINGS` can be set to `default`, in other words, `PYTHONWARNINGS=default`.
```

mauritsvanrees commented on docs/developer-guide/deprecation.md:
````diff
@@ -0,0 +1,265 @@
+    moved("new.pkg.bar", "Version 2.0")
+    ```
+
+1.  Now you can still import the namespace from `bar` at the old place, but get a deprecation warning.
````
This confuses me a bit.

```suggestion
1.  Now you can still import the module at the old place, but get a deprecation warning.
```

mauritsvanrees commented on docs/developer-guide/deprecation.md:
```diff
@@ -0,0 +1,265 @@
+Sometimes you need to move templates to new locations.
+Since add-ons often use [`z3c.jbot`](https://github.com/zopefoundation/z3c.jbot) to override templates by their position, you'll need to point them to the new position as well as make sure that the override still works with the old position.
+
+To deprecate a package, follow these steps.
```
```suggestion
To deprecate a template, follow these steps.
```

mauritsvanrees commented on docs/developer-guide/deprecation.md:
```diff
@@ -0,0 +1,265 @@
+        >
+    
+      <include
+          zcml:condition="installed z3c.jbot"
+          package="z3c.jbot"
```
We should explicitly load `meta.zcml` of z3c.jbot`:

```suggestion
      <include
          package="z3c.jbot"
          file="meta.zcml"
          zcml:condition="installed z3c.jbot"
```

That is the file that defines the `jbot-deprecations` feature and the `jbotDeprecated` directive.
This is how [`plone.locking` does it](https://github.com/plone/plone.locking/blob/4.0.0a3/src/plone/locking/configure.zcml#L14-L23).

mauritsvanrees reviewed (changes requested):
Thanks for working on this.  I added some suggestions.

jensens committed:
Update docs/developer-guide/deprecation.md

Co-authored-by: Maurits van Rees <maurits@vanrees.org>

jensens committed:
Update docs/developer-guide/deprecation.md

Co-authored-by: Maurits van Rees <maurits@vanrees.org>

jensens committed:
Update docs/developer-guide/deprecation.md

Co-authored-by: Maurits van Rees <maurits@vanrees.org>

jensens committed:
Update docs/developer-guide/deprecation.md

Co-authored-by: Maurits van Rees <maurits@vanrees.org>

jensens committed:
Merge branch '6.0' into chapter-deprecation

stevepiercy commented on docs/developer-guide/deprecation.md:
```diff
@@ -0,0 +1,266 @@
+
+To enable deprecation warnings, use the `-W` command.
+
+Given you're using a modern buildout with a virtual environment as recommended, the command would be the following
```
```suggestion
Given you're using a modern buildout with a virtual environment as recommended, the command would be the following.
```

stevepiercy commented on docs/developer-guide/deprecation.md:
```diff
@@ -0,0 +1,266 @@
+
+:   Warnings are written to `stderr` by default, but `DeprecationWarning` output is surpressed by default.
+    
+    Output can be enabled by starting the Python interpreter with the {ref}`-W[all|module|once]  <python:using-on-warnings>` argument.
```
```suggestion
    Output can be enabled by starting the Python interpreter with the {ref}`-W[all|module|once] <python:using-on-warnings>` argument.
```

stevepiercy commented on docs/developer-guide/deprecation.md:
```diff
@@ -0,0 +1,266 @@
+
+## Enable deprecation warnings
+
+This section describes how to enable deprecation warnings in Zope, Python, and tests.
```
```suggestion
This section describes how to enable deprecation warnings in Python, both in an interpreter and code, and when running tests.
```

stevepiercy commented on docs/developer-guide/deprecation.md:
````diff
@@ -0,0 +1,266 @@
+    action always
+    category exceptions.DeprecationWarning
+</warnfilter>
+```
````
Per @mauritsvanrees's suggestion.

```suggestion
```

stevepiercy reviewed (changes requested):
A couple of minor tweaks and the removal of the Zope section and revision of the parent heading content.

jensens committed:
Update docs/developer-guide/deprecation.md

Co-authored-by: Steve Piercy <web@stevepiercy.com>

jensens committed:
Update docs/developer-guide/deprecation.md

Co-authored-by: Steve Piercy <web@stevepiercy.com>

jensens committed:
Update docs/developer-guide/deprecation.md

Co-authored-by: Steve Piercy <web@stevepiercy.com>

jensens committed:
Update docs/developer-guide/deprecation.md

Co-authored-by: Steve Piercy <web@stevepiercy.com>

jensens committed:
Merge branch '6.0' into chapter-deprecation

gforcada commented on docs/developer-guide/deprecation.md:
````diff
@@ -0,0 +1,234 @@
+    )
+```
+
+### Deprecate a template position
````
```suggestion
### Deprecate a template location
```

stevepiercy replied:
There's also a mixture of "position" and "location" in the narrative text. Does a template have a position or a location? Whichever is more appropriate, we should use in the narrative text as well.

For the record, location pertains to file system, whereas position pertains to an order within a group of similar items.

stevepiercy replied:
@plone/classicui-team please respond. Does a template have a "position" or a "location"? I'll update the text accordingly. Let's get this merged! Thank you!

stevepiercy committed:
Apply suggestion from @gforcada

Co-authored-by: Gil Forcada Codinachs <gil.gnome@gmail.com>

stevepiercy committed:
s/position/location

stevepiercy committed:
Merge branch '6.0' into chapter-deprecation

stevepiercy commented:
@mauritsvanrees @gforcada @jensens @ale-rt would you please do one final review of this chapter? I don't have an approval from someone who knows this stuff. Thank you!

ale-rt commented on docs/developer-guide/deprecation.md:
````diff
@@ -0,0 +1,234 @@
+1.  Add this deprecation snippet to the package {file}`configure.zcml` file.
+
+    ```{code-block} xml
+    :emphasize-lines: 7-14
````
```suggestion
    :emphasize-lines: 7-15
```

ale-rt commented on docs/conceptual-guides/index.md:
```diff
@@ -16,8 +16,9 @@ This part of the documentation provides explanation of concepts to deepen and br
 choose-user-interface
-compare-buildout-pip
 distributions
+compare-buildout-pip
```
I have no idea if this is intentional or a merge artifact

stevepiercy replied:
Good catch. It was a merge artifact. Thank you!

stevepiercy replied:
Fixed in https://github.com/plone/documentation/commit/8dfd489c5d631b0d881be1c6f80d130fc9486b03

ale-rt reviewed (dismissed):
This is amazing.
I would merge and if there are some small changes needed to fine tune this one we can do smaller incremental PRs.

stevepiercy committed:
Fix order of items in the table of contents

stevepiercy committed:
Include one more line to emphasize

Co-authored-by: Alessandro Pisa <alessandro.pisa@gmail.com>

stevepiercy commented:
Almost one year later, we finally have an official deprecation policy and guide for Plone 6.0, just in time for 6.2. We move at the speed of open source volunteers. 🐌  Merging!

davisagli commented:
Nice work!
